spec/models/zip_code_spec.rb

```diff
@@ -3,5 +3,75 @@
 require 'rails_helper'
 
 RSpec.describe ZipCode, type: :model do
-  it { should validate_presence_of(:d_codigo) }
+  context 'validations tests' do
+    it { should validate_presence_of(:d_codigo) }
+  end
+
+  context 'scopes tests' do
+    let(:params) do
+      { d_codigo: '01000',
+        d_asenta: 'San Ángel',
+        d_tipo_asenta: 'Colonia',
+        d_mnpio: 'Álvaro Obregón',
+        d_estado: 'Ciudad de México',
+        d_ciudad: 'Ciudad de México',
+        d_cp: '01001',
+        c_estado: '09',
+        c_oficina: '01001',
+        c_cp: '.',
+        c_tipo_asenta: '09',
+        c_mnpio: '010',
+        id_asenta_cpcons: '0001',
+        d_zona: 'Urbano',
+        c_cve_ciudad: '01' }
+    end
+
+    before(:each) do
```

Comment:
It is better if you move this into a `let` directive to lazy load the record, although it may not be of much help, but it will make the code more clean

Reply:
I prefer to use `FactoryBot.create(:zip_code)`

Reply:
Yes, you actually should use both, just to clarify:
:point_right: `let` is only for **RSpec** and allows you lazy load any model regardless of its way of creation, whether is from `FactoryBot` or `ZipCode.create`, or any other way
:point_right: `FactoryBot` is just a factory pattern implementation to easily build or create models, but has nothing to do with RSpec, although they are commonly use together.

So in this case you could refactor this to something like:

```ruby
let(:zip_code) { FactoryBot.create(:zip_code) }
```

Reply:
Done!